tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -784,4 +787,36 @@ internal static IastModuleResponse OnXpathInjection(string xpath)
             return IastModuleResponse.Empty;
         }
     }
+
+    internal static void OnEmailHtmlInjection(object? message)
+    {
+        if (!Iast.Instance.Settings.Enabled)
+        {
+            return;
+        }
+
+        try
+        {
+            OnExecutedSinkTelemetry(IastInstrumentedSinks.EmailHtmlInjection);
+
+            if (message is null)
+            {
+                return;
+            }
+
+            var messageDuck = message.DuckCast<IMailMessage>();
+
+            if (messageDuck?.IsBodyHtml is not true || string.IsNullOrEmpty(messageDuck.Body))
+            {
+                return;
+            }
+
+            // We use the same secure marks as XSS
+            GetScope(messageDuck.Body, IntegrationId.EmailHtmlInjection, VulnerabilityTypeName.EmailHtmlInjection, OperationNameEmailHtmlInjection, taintValidator: Always, exclusionSecureMarks: SecureMarks.Xss);
+        }
+        catch (Exception ex)
```

Comment:
wouldn't this catch `BlockException` ? 
also given the Send method does:
```   public static object? Send(object? message)
    {
        try
        {
            IastModule.OnEmailHtmlInjection(message);
            return message;
        }
        catch (Exception ex) when (ex is not BlockException)
        {
            IastModule.Log.Error(ex, $"Error invoking {nameof(SmtpClientAspect)}.{nameof(Send)}");
            return message;
        }
    }
```
isnt this catch redundant? or should the caller's one be removed (in Send)?

Reply:
A new analyzer was added today that forces us to try/catch in the aspects with that pattern and was added in the merge (the one that you mention in your comment). That made this try/catch obsolete. What is interesting is that other vulnerabilities that followed the previous pattern should also remove the try/catch blocks that are redundant with the PR that was added today. Thanks for spotting that one.